### DISCUSSION TITLE
Which cloud-based tool supports parallel testing using detox ?

### DISCUSSION THREAD

**Discussion by Rashivag** [wix/Detox#3096] [Q&A]
Planning to run parallel tests on multiple android virtual devices for a mobile application (developed in react-native). Already have a  detox test suite for the same application. Would like to know which tool (supporting detox)will be most helpful for parallel testing, few of the options that I know:
1. SauceLabs
2. BrowserStack
3. AWS device farm
4. Bitbar

**Comment by noomorph** [comment]
At the moment, we're scaling with Genymotion Cloud: https://medium.com/genymobile/scale-your-e2e-react-native-ui-testing-with-detox-on-genymotion-cloud-android-virtual-devices-b3effbc7eccc . Purely from the technical point, it is the most explored (paved) way by us.

However, with Detox 20, I hope to see more flexibility with implementing provider-specific device setups. The current Detox 19 is still a bit rigid in this sense, although it *is* expected to work with anything that can provide you a device over ADB, provided you write wrappers (or shell scripts) that handle the allocation/deallocation.

Mentioning @d4vidi for visibility.

**Reply by Rashivag** [reply]
@noomorph Thanks a lot for the prompt reply.
May I know when is the Detox 20 release planned ? It will be helpful for us to make a decision.

**Reply by noomorph** [reply]
Initially we planned to release it this quarter but apparently we're delaying. Now it seems more realistic  to happen around January, or mid-February in the worst case.

**Comment by pke** [comment]
BrowserStack claims to work on support since 2018... nothing in sight. Would be nice they would try again with Detox 20.
https://twitter.com/browserstack/status/970592634829000704?s=20&t=T800QF3-RChFUbpPj54NZA

![image](https://user-images.githubusercontent.com/164625/154708161-ff9b868c-feb7-4e9b-baf3-029725291bf0.png)
